lib/modules/manager/devcontainer/extract.ts

```diff
@@ -0,0 +1,103 @@
+import { logger } from '../../../logger';
+import { isValidDependency } from '../custom/regex/utils';
+import { getDep } from '../dockerfile/extract';
+import type {
+  ExtractConfig,
+  PackageDependency,
+  PackageFileContent,
+} from '../types';
+import { DevContainerFile } from './schema';
+
+export function extractPackageFile(
+  content: string,
+  packageFile: string,
+  extractConfig: ExtractConfig,
+): PackageFileContent | null {
+  try {
+    const file = DevContainerFile.parse(content);
+    let targets: string[] = [];
+    const image = getImage(file);
+
+    if (image !== undefined && image !== null) {
+      targets.push(image);
+    } else {
+      logger.debug(
```

Comment:
```suggestion
      logger.trace(
```
no need to be so verbose